bin/registerServerTopology-Func.php

```diff
@@ -125,18 +138,85 @@ function getChildren(CentreonDB $db): array
  */
 function registerCentralCredentials(CentreonDB $db, array $loginCredentials): void
 {
-    $query = "INSERT INTO `informations` (`key`, `value`) VALUES ('apiUsername', :username), ('apiCredentials', :pwd)";
+    $queryValue = '';
+    $count = 1;
+    $bindValues = [];
+    $proxyCredentials = $loginCredentials['proxy_informations'] ?? [];
+    $loginCredentials = array_filter($loginCredentials, function ($key) {
+        return $key !== 'proxy_informations';
+    }, ARRAY_FILTER_USE_KEY);
+    foreach ($loginCredentials as $key => $value) {
+        if ($count === count($loginCredentials)) {
+            $queryValue .= " ('$key', :$key)";
+        } else {
+            $queryValue .= " ('$key', :$key), ";
+        }
+        $count++;
+        switch ($key) {
+            case 'apiPort':
+                $bindValues[':' . $key] = [
+                    \PDO::PARAM_INT => $value
+                ];
+                break;
+            default:
+                $bindValues[':' . $key] = [
+                    \PDO::PARAM_STR => $value
+                ];
+                break;
+        }
+    }
+    $db->query("DELETE FROM informations WHERE `key` LIKE 'api%'");
+    $query = "INSERT INTO `informations` (`key`, `value`) VALUES $queryValue";
     $statement = $db->prepare($query);
-    $statement->bindValue(':username', $loginCredentials['login'], \PDO::PARAM_STR);
-    $statement->bindValue(':pwd', $loginCredentials['password'], \PDO::PARAM_STR);
+    foreach ($bindValues as $token => $bindParams) {
+        foreach ($bindParams as $paramType => $paramValue) {
+            $statement->bindValue($token, $paramValue, $paramType);
+        }
+    }
     $statement->execute();
+
+    if (!empty($proxyCredentials)) {
+        $queryValue = '';
+        $count = 1;
+        $bindProxyValues = [];
+        foreach ($proxyCredentials as $key => $value) {
+            if ($count === count($proxyCredentials)) {
+                $queryValue .= " ('$key', :$key)";
+            } else {
+                $queryValue .= " ('$key', :$key), ";
+            }
+            $count++;
+            switch ($key) {
+                case 'proxy_port':
+                    $bindProxyValues[':' . $key] = [
+                        \PDO::PARAM_INT => $value
+                    ];
+                    break;
+                default:
+                    $bindProxyValues[':' . $key] = [
+                        \PDO::PARAM_STR => $value
+                    ];
+                    break;
+            }
+        }
+
+        $db->query("DELETE FROM options WHERE `key` LIKE 'proxy_%'");
+        $query = "INSERT INTO `options` (`key`, `value`) VALUES $queryValue";
+
+        $statement = $db->prepare($query);
+        foreach ($bindProxyValues as $token => $bindParams) {
+            foreach ($bindParams as $paramType => $paramValue) {
+                $statement->bindValue($token, $paramValue, $paramType);
+            }
+        }
+        $statement->execute();
```

Comment:
Please use a transaction to cancel all changes in case of errors.

Reply:
done l.96

Reply:
done l.96